Bugsnag/Client/BugsnagClient.m

```diff
@@ -636,6 +619,72 @@ - (void)start {
     [self.metadata addMetadata:BSGParseDeviceMetadata(@{@"system": systemInfo}) toSection:BSGKeyDevice];
 }
 
+- (bool)shouldReportOOM {
+#if BSGOOMAvailable
+    // Disable if in an app extension, since app extensions have a different
+    // app lifecycle and the heuristic used for finding app terminations rooted
+    // in fixable code does not apply
+    if([BSG_KSSystemInfo isRunningInAppExtension]) {
+        return NO;
+    }
+
+    // autoDetectErrors disables all unhandled event reporting
+    if(!self.configuration.autoDetectErrors) {
+        return NO;
+    }
+
+    // Are OOMs enabled?
+    if(!self.configuration.enabledErrorTypes.ooms) {
+        return NO;
+    }
+
+    return [self didLikelyOOM];
+#else
+    return NO;
+#endif
+}
+
+/**
+ * These heuristics aren't 100% guaranteed to be correct, but they're correct often enough to be useful.
+ */
+- (bool)didLikelyOOM {
```

Comment:
Should we move this method into BugsnagSystemState?

It would help reduce the size of BugsnagClient a bit, and the method doesn't access anything in the client.

Reply:
I'm trying to keep BugsnagSystemState's purpose to only recording information rather than interpreting it.

Of course, BugsnagClient isn't really the ideal place to interpret it either I suppose...